docs: 📚 Document used typographic styles and fonts (#617)
synergy-design-system/synergy-design-system: packages/docs/stories/Prerequisites.mdx

import { Meta, Markdown } from "@storybook/blocks";
import Contributing from '../../../CONTRIBUTING.md?raw';

<Meta title="Prerequisites" />

# Prerequisites

## Fonts

The Synergy Design System makes use of the `Open Sans` typeface.
Because there are various ways of loading fonts, depending on used bundler, pre- and postprocessors and usage of a CDN, Synergy does **not provide** this font.

> You will need to include it in your project.

Reviewer: ```suggestion
You need to include it into your own project.
```

The following information may be helpful to get you started including your own copy of `Open Sans` in your project for quicker bootstrapping:

### Local Installation

> 1. Download the `Open Sans` font from the [SICK Brand Portal](https://brand.sick.com/document/49#/-/resources-1).

Reviewer: Why not referencing the fonts on GitHub (packages/docs/public/OpenSans) ?
I'd prefer not to reference servers we don't have control over.

Author: As the brand portal is the single source of truth, we will leave the comment as discussed.

Reviewer 2: Google also only provides .ttf files in the official GitHub

Author: Another valid point. I think there was the possibility to load the `woff` files somewhere, but I did not find it yesterday, too.

Reviewer 2: Yes, I also can't remember where I have downloaded the woff files initially.
Maybe buried somwhere in the older version of Google Fonts.

> 2. Extract the ZIP file to a destination reachable by your project (e.g. a `public` folder).

Reviewer: this step can be skipped then

1. Install the `@fontsource/open-sans` package into your project.
2. Include the font in your project by adding the following import statements to your project:

```js
import '@fontsource/open-sans/400.css';
import '@fontsource/open-sans/400-italic.css';
import '@fontsource/open-sans/600.css';
import '@fontsource/open-sans/600-italic.css';
import '@fontsource/open-sans/700.css';
import '@fontsource/open-sans/700-italic.css';
```

### Installing via local npm package (angular)

1. Install the `@fontsource/open-sans` package into your project.
2. Adjust your `angular.json` file to include the needed stylesheets (where project_name is the name of your project):

```json
{
  "projects": {
    "project_name": {
      "architect": {
        "build": {
          "options": {
            "styles": [
              "@fontsource/open-sans/400.css",
              "@fontsource/open-sans/400-italic.css",
              "@fontsource/open-sans/600.css",
              "@fontsource/open-sans/600-italic.css",
              "@fontsource/open-sans/700.css",
              "@fontsource/open-sans/700-italic.css",
            ]
          }
        }
      }
    }
  }
}
```

### Installing via Google Fonts

> > ⚠️ This is only suited for development purposes and **must never be used in production**.

Reviewer: Could we add the "why" ? Why is only suitable for development? Why should I use this code in the first place?

Author: see above

1. Install the `@fontsource/open-sans` package into your project.
2. Include the font in your project by adding the following import statements to your project:

```js
import '@fontsource/open-sans/400.css';
import '@fontsource/open-sans/400-italic.css';
import '@fontsource/open-sans/600.css';
import '@fontsource/open-sans/600-italic.css';
import '@fontsource/open-sans/700.css';
import '@fontsource/open-sans/700-italic.css';
```

### Installing via local npm package (angular)

1. Install the `@fontsource/open-sans` package into your project.
2. Adjust your `angular.json` file to include the needed stylesheets (where project_name is the name of your project):

```json
{
  "projects": {
    "project_name": {
      "architect": {
        "build": {
          "options": {
            "styles": [
              "@fontsource/open-sans/400.css",
              "@fontsource/open-sans/400-italic.css",
              "@fontsource/open-sans/600.css",
              "@fontsource/open-sans/600-italic.css",
              "@fontsource/open-sans/700.css",
              "@fontsource/open-sans/700-italic.css",
            ]
          }
        }
      }
    }
  }
}
```

### Using the SICK CDN

> This is the simplest default font behavor. The `Open Sans` font is loaded from the SICK CDN.

Reviewer 3: typo `behavor` --> `behavior`

```suggestion
This is the simplest default font behavior. The `Open Sans` font is loaded from the SICK CDN.
```
